### PR TITLE
[11.0][ADD]stock_push_delay v11

### PR DESCRIPTION
**Pull request by AaronHForgeFlow**
Some times we done want to apply push rules right away. For example, when receiving partial shipments in several steps, we want to trigger manually the picking form the packing zone to Input and to Stock. That helps to track better quantities on chained moves, because in many use cases, the pickings are not completed entirely in one step, and specific actions are required depending on the receipt location, among other reasons.

@ForgeFlow

### CODE REVIEW

File: stock_push_delay/models/stock.py [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+class StockPicking(models.Model):
+
+    _inherit = 'stock.picking'
+
+    @api.multi
+    def action_done(self):
+        res = super(StockPicking, self).action_done()
+        for picking in self:
+            picking.move_lines.with_context(manual_push=True)._push_apply()
```
Comment (rousseldenis) [inline stock_push_delay/models/stock.py]: Don't do that with context, so bad for performances.

Either you create another method (e.g.: _push_apply_manual()) , either you put context on self.

**Review by rousseldenis — CHANGES_REQUESTED** [review]

**Review by rousseldenis — APPROVED** [review]
Can you squash ?

File: stock_push_delay/models/stock.py (hunk 1 above)
Comment (AaronHForgeFlow) [reply stock_push_delay/models/stock.py]: Thanks for the comment, did not realize about that performance issue.

**Review by JordiBForgeFlow — APPROVED** [review]

**Review by AdriaGForgeFlow — APPROVED** [review]

**Pull request merged**